cpp/src/io/parquet/parquet_gpu.hpp

```diff
@@ -520,11 +541,13 @@ constexpr bool is_string_col(ColumnChunkDesc const& chunk)
  * @brief Launches kernel for parsing the page headers in the column chunks
  *
  * @param[in] chunks List of column chunks
+ * @param[in] chunk_pages List of pages associated with the chunks, in chunk-sorted order
  * @param[in] num_chunks Number of column chunks
  * @param[out] error_code Error code for kernel failures
  * @param[in] stream CUDA stream to use
  */
 void DecodePageHeaders(ColumnChunkDesc* chunks,
+                       chunk_page_info* chunk_pages,
```

Comment:
Pointer of pointer? Can we just use `PageInfo*`?

Reply:
This an array of arrays.  Ie, every element of in chunk_pages[] is a separate array of pages.  That is, each chunk has an array of pages relevant to it.  It's (the PageInfo*) not stored as a span because the size of the arrays is stored in the chunks themselves:

https://github.com/rapidsai/cudf/blob/b01b06e4a17be55e009e1eabde09c07f767bacba/cpp/src/io/parquet/page_hdr.cu#L397

https://github.com/rapidsai/cudf/blob/b01b06e4a17be55e009e1eabde09c07f767bacba/cpp/src/io/parquet/page_hdr.cu#L458